File: ONLINE_CODING_TEST/school.programmers/lvl_1/12910_나누어_떨어지는_숫자_배열.js
function solution(arr, divisor) {
  var answer = [];

  answer = arr.filter(ele => ele % divisor === 0).sort((a, b) => a - b);

  // var limit = arr.length;
  // for (var i = 0; i <= limit; i ++) {
  //     if (arr[i] % divisor === 0) answer.push(arr[i]);
  // }

  return answer.length >= 1 ? answer : [-1];
}

const arrArr = [{
    arr: [5, 9, 7, 10],
    divisor: 5
  },
  {
    arr: [2, 36, 1, 3],
    divisor: 1
  },
  {
    arr: [3, 2, 6],
    divisor: 10
  }
];

for (let obj of arrArr) {
  console.log(obj, solution(obj.arr, obj.divisor));
}

File: ONLINE_CODING_TEST/school.programmers/lvl_1/12910_나누어_떨어지는_숫자_배열.java
import java.util.ArrayList;
import java.util.Collections;
class Solution {
    public int[] solution(int[] arr, int divisor) {
        int[] answer = null;
        ArrayList<Integer> list = new ArrayList<>();
        for (int i : arr) {
            if (i % divisor == 0) list.add(i);
        }
        
        if (list.size() == 0 || list.isEmpty()) {
            answer = new int[]{-1};
            return answer;
        }
        
        Collections.sort(list);
        answer = new int[list.size()];
        int idx = 0;
        for (int i : list) {
            answer[idx++] = i;
        }
        return answer;
    }
}